src/task.rs

```diff
@@ -100,15 +713,72 @@ struct State {
     waker: AtomicWaker,
 }
 
-impl TaskMetrics {
-    pub fn new() -> TaskMetrics {
-        TaskMetrics::with_slow_poll_cut_off(Duration::from_micros(10))
+impl Monitor {
+
+    /// The default duration at which polls cross the threshold into being categorized as 'slow' is 50μs.
+    #[cfg(not(test))]
+    pub const DEFAULT_SLOW_POLL_THRESHOLD: Duration = Duration::from_micros(50);
+    #[cfg(test)]
+    pub const DEFAULT_SLOW_POLL_THRESHOLD: Duration = Duration::from_millis(500);
+
+    /// Constructs a new task monitor.
+    /// 
+    /// Uses [`Self::DEFAULT_SLOW_POLL_THRESHOLD`] as the threshold at which polls will be considered 'slow'.
+    pub fn new() -> Monitor {
+        Monitor::with_slow_poll_threshold(Self::DEFAULT_SLOW_POLL_THRESHOLD)
     }
 
-    pub fn with_slow_poll_cut_off(slow_poll_cut_off: Duration) -> TaskMetrics {
-        TaskMetrics {
-            metrics: Arc::new(Metrics {
-                slow_poll_cut_off,
+    /// Constructs a new task monitor with a given threshold at which polls are considered 'slow'.
+    /// 
+    /// ##### Selecting an appropriate threshold
+    /// TODO. What advice can we give here?
```

Comment:
Task poll time is not a primary metric, but a metric used to debug issues such as slow response times. The ideal threshold will be application-specific based on your tolerances and the scheduler queue depth, and probably between 10 and 75 microseconds. For example, if the scheduler peaks at 100 tasks waiting in the scheduled queue and the mean task poll time is 100 microseconds, then the scheduling delay will be 100 * 100us, or 10 milliseconds. If a request requires 5 `.await` calls to complete, then it will spend 50 milliseconds just waiting to be scheduled, which is fairly high.